flycheck.el

```diff
@@ -6976,6 +6976,36 @@ Otherwise return the previously used cache directory."
         (or flycheck-haskell-ghc-cache-directory
             (make-temp-file "flycheck-haskell-ghc-cache" 'directory))))
 
+(defun flycheck-haskell--default-directory (checker)
+  "Come up with a sutable default directory for Haskell to run checker in."
```

Comment:
Come up sounds strange. Why not simply "find"?

Please also extend the docstring to explain what a "suitable" directory would be, i.e. document the behaviour of the function as well.